test/e2e/yarn-pnp/test/utils.ts

```diff
@@ -0,0 +1,49 @@
+import fs from 'fs-extra'
+import { join } from 'path'
+import { fetchViaHTTP } from 'next-test-utils'
+import { createNext, FileRef } from 'e2e-utils'
+import { NextInstance } from 'test/lib/next-modes/base'
+
+jest.setTimeout(2 * 60 * 1000)
+
+export function runTests(example = '') {
+  let next: NextInstance
+
+  beforeAll(async () => {
+    const srcDir = join(__dirname, '../../../../examples', example)
+    const srcFiles = await fs.readdir(srcDir)
+
+    const packageJson = await fs.readJson(join(srcDir, 'package.json'))
+
+    next = await createNext({
+      files: srcFiles.reduce((prev, file) => {
+        if (file !== 'package.json') {
+          prev[file] = new FileRef(join(srcDir, file))
+        }
+        return prev
+      }, {} as { [key: string]: FileRef }),
+      dependencies: {
+        ...packageJson.dependencies,
+        ...packageJson.devDependencies,
+      },
+      installCommand: ({ dependencies }) => {
+        const pkgs = Object.keys(dependencies).reduce((prev, cur) => {
+          prev.push(`${cur}@${dependencies[cur]}`)
+          return prev
+        }, [] as string[])
+        return `yarn set version berry && yarn config set enableGlobalCache true && yarn config set compressionLevel 0 && yarn add ${pkgs.join(
```

Comment:
Should we add more test with different combinations of these settings? For example `set enableGlobalCache false`?

Reply:
I think this should still behave the same from Next.js' point of view and we would be more so testing yarn itself with different config values. The global caching is to help speed up the installs for tests since it seems installing with PnP is slower making the tests take a bit.